core/src/main/java/org/testcontainers/containers/ComposeContainer.java

```diff
@@ -61,36 +62,74 @@ public class ComposeContainer extends FailureDetectingExternalResource implement
+    private static final String DEFAULT_DOCKER_IMAGE = "docker:27.5.0";
 
     private final ComposeDelegate composeDelegate;
 
     private String project;
 
     private List<String> filesInDirectory = new ArrayList<>();
 
+    public ComposeContainer(DockerImageName image, File... composeFiles) {
+        this(image, Arrays.asList(composeFiles));
+    }
+
+    public ComposeContainer(DockerImageName image, List<File> composeFiles) {
+        this(image, Base58.randomString(6).toLowerCase(), composeFiles);
+    }
+
+    public ComposeContainer(DockerImageName image, String identifier, File... composeFiles) {
+        this(image, identifier, Arrays.asList(composeFiles));
+    }
+
+    public ComposeContainer(DockerImageName image, String identifier, List<File> composeFiles) {
+        this.composeDelegate =
+            new ComposeDelegate(ComposeDelegate.ComposeVersion.V2, composeFiles, identifier, COMPOSE_EXECUTABLE, image);
+        this.project = this.composeDelegate.getProject();
+    }
+
+    /**
+     * @deprecated
+     *  Use the new constructor ComposeContainer(DockerImageName image, File... composeFiles)
+     */
+    @Deprecated
     public ComposeContainer(File... composeFiles) {
-        this(Arrays.asList(composeFiles));
+        this(getDockerImageName(), Arrays.asList(composeFiles));
     }
 
+    /**
+     * @deprecated
+     *  Use the new constructor ComposeContainer(DockerImageName image,List composeFiles)
+     */
+    @Deprecated
     public ComposeContainer(List<File> composeFiles) {
-        this(Base58.randomString(6).toLowerCase(), composeFiles);
+        this(getDockerImageName(), composeFiles);
     }
 
+    /**
+     * @deprecated
+     *  Use the new constructor ComposeContainer(DockerImageName image, String identifier, File... composeFile)
+     */
+    @Deprecated
     public ComposeContainer(String identifier, File... composeFiles) {
-        this(identifier, Arrays.asList(composeFiles));
+        this(getDockerImageName(), identifier, Arrays.asList(composeFiles));
     }
 
+    /**
+     * @deprecated
+     * Use the new constructor ComposeContainer(DockerImageName image,String identifier, List composeFiles)
+     */
+    @Deprecated
     public ComposeContainer(String identifier, List<File> composeFiles) {
-        this.composeDelegate =
-            new ComposeDelegate(
-                ComposeDelegate.ComposeVersion.V2,
-                composeFiles,
-                identifier,
-                COMPOSE_EXECUTABLE,
-                DEFAULT_IMAGE_NAME
-            );
-        this.project = this.composeDelegate.getProject();
+        this(getDockerImageName(), identifier, composeFiles);
+    }
+
+    public static DockerImageName getDockerImageName() {
+        return DockerImageName.parse(
+            TestcontainersConfiguration
+                .getInstance()
+                .getEnvVarOrUserProperty("compose.container.image", DEFAULT_DOCKER_IMAGE)
+        );
```

Comment:
This is a deprecated way to do it and given we are offering a new constructor to set the image, this doesn't apply anymore.